### ISSUE TITLE
LG 37LH4000 - ChanSort edited lists screw up TV completely

### ISSUE THREAD

**Issue by MesserJo-BB-GH** [PredatH0r/ChanSort#61]
I used ChanSort for the subject TV successfully for some time. However, last time it destroyed the TV completely after importing a new edited list. The TV error is: "This function is not available now" when pressing a number key or channel up or down. LG support says and in the LG user forum it is stated, that the main board is damaged, which is not true. I managed to reset the TV to factory default, then did a channel search and it worked as before. Exporting this new channel list, sorting/editing with ChanSort and reimporting screwed up the TV again. I'm using the latest software version, and the TV uses the latest LG firmware version 3.75. Any idea what is going wrong?

**Comment by Geekgasm** [comment]
I'm also experiencing problems with importing the channel list into my LG tv after editing it with ChanSort. Since the symptoms look quite differently in my case, I opened a new issue #64 for this.
But maybe it is related?

**Comment by PredatH0r** [comment]
To investigate the issue I would need the unmodified and the modified version of your channel list.
Also, try importing the unmodified file right after exporting. Some LG firmware versions fail to read their own exported files.

**Comment by MesserJo-BB-GH** [comment]
Sorry for the late answer. Our cable company did a resorting of channels, therefore I needed to scan all, stored the list and uploaded it again. The result is again: screwed up channels. Initially evreything looks good, but after switching "off" and "on" the TV, the channels are completely wrong sorted. Here the results: [https://www.dropbox.com/sh/vf29kcfj5gcnmta/AAAe6XBJKLNf0ho2DynTfxZNa?dl=0](https://www.dropbox.com/sh/vf29kcfj5gcnmta/AAAe6XBJKLNf0ho2DynTfxZNa?dl=0)
The unmodified list is imported correctly. But saving and reopening the same, modified file resulted in wrong order of channels if I "append not sorted channels", if I chose "delete unsorted channels" it works better, but doesn't solve my issue.

**Comment by PredatH0r** [comment]
When you export the channel list from your TV, does it create 2 TLL files on the stick (GlobalClone and one specific to your TV model)?
ChanSort only edits the GlobalClone file. I do not know if your TV also reads the other file too when importing the last back. The behaviour seems to be different between models and firmware versions.
To my knowledge it's best to delete the non-GlobalClone TLL file from the stick to avoid inconsistent data during the import.

I used text diffing tools to compare the original files with those modified by ChanSort. The only changes are literally the channel numbers and the indicator, that the number was changed by the user.
If the TV goes nuts by reading such a file, then I can only blame the TV's firmware.

I'm also not sure how the TV handles channel numbers when your cable provider sends out his preferred numbers (LCN). I know from other TV manufacturers that they do not allow the user to change numbers that were assigned by the cable company.
With older LG TVs it depended on the options taken during the TV setup, whether you select a specific cable provider or a neutral setting like "none", "other", "..." to prevent using predefined numbers. I can't tell if that still matters with newer models.

**Comment by MesserJo-BB-GH** [comment]
No, there is only one file *.tll created during export. When I import the modified file, I can select the one I created on my PC. There is no second file (Global). What I now also tried is to have only one TLL file on the USB-Stick. Unfortunately no change. 

Sorry, but I am giving up on thie old LG TV. 

BTW: For two other TVs (Sony and Samsung) ChanSort doesn't work at all, but both companies offer their own (poorly designed) software. Usability is a mess, but the result works.

**Comment by PredatH0r** [comment]
Could you load your "after_LG_search" list into the TV, swap channels 1 and 2 through the TV menu, export that list again and send it to me?
It would be very helpful for me to see what changes the TV made to the file and make sure ChanSort acts the same way.

**Comment by PredatH0r** [comment]
I also just noticed that you set favorites with ChanSort, which could also be the source of the problem.
When you export the after_LG_search list (as I asked in the previous message), can you also also please set the 3rd channel as Fav1 and 4th channel as Fav2 through your TV menu?
That should make clear which fields the TV is using for that information.

**Comment by PredatH0r** [comment]
If you TV's service menu has a setting "DTV channel update" make sure it is set to disabled.
I don't know exactly under which circumstances this is needed (cable-TV provider, TV model, firmware version), but I recall a time when I had to use that setting on my 2012 LM420 model. If the TV tries to auto-update a modified list, it might screw things up.

**Comment by MesserJo-BB-GH** [comment]
Sorry for the delay, but I was busy with a lot of other things. Now I uploaded four TLL files to the same DropBox folder: https://www.dropbox.com/sh/vf29kcfj5gcnmta/AAAe6XBJKLNf0ho2DynTfxZNa?dl=0

xx-DVB-EU00001.TLL = the working channel list with my favorites
xx-DVB-EU00002.TLL = the list after completed search, no changes made by myself
xx-DVB-EU00003.TLL = after switching channel 1 and 2 
xx-DVB-EU00004.TLL = set channel 3 to favorite 1  and channel 3 to favorite 2

On my TV I couldn't find the setting "DTV channel update".

I hope that you can evaluate the problem with this additional information.
